### ISSUE TITLE
Job fire and forget is not working

### ISSUE THREAD

**Issue by johnwc** [elsa-workflows/elsa-core#3161]
I created a master workflow that has several other workflows chained one after another. Each start workflow activity is set to `Fire and forget`. But, the workflows only start after the previous one in the chain completes.

**Comment by sfmskywalker** [comment]
Yeah, the "fire and forget" term is misleading in this context. What the **Run Workflow** activity actually does is execute a workflow and *wait for it to either complete or get suspended*. If the child workflow gets suspended, then the "fire and forget" setting will cause the main workflow to continue to its next activity. The "blocking" mode on the other hand will suspend itself until the child workflow completes.

What you're after is a true fire and forget behavior, where the main workflow dispatches the child workflow, not execute it.
We should probably add a new mode called e.g. "TRUE fire and forget" (maybe using a better name 😉).

The following would be a proper list of modes:

- Blocking (waits for the child workflow to complete)
- Start (starts the child workflow until it either completes or gets suspended)
- Fire and forget (dispatches the child workflow and carries on to the next activity)

**Comment by johnwc** [comment]
After much trial and error, I ended up stumbling upon adding a short 10 second timer to the workflows we are starting so that they suspend and allow the parent to continue. So this will suffice for now until there is an official solution.

I think the `Start` need to be so that it Starts the workflow but waits for it to complete none blocking. To me `Fire and Forget` tells me that the workflow is started and we don't care the outcome at all and will not wait for anything, we are just initiating it and moving on.

**Comment by Adrian-Grimm** [comment]
Would be nice to get a real fire and forget - which is also capable to spread the instanced distributed across all workers =) .

**Comment by sfmskywalker** [comment]
Exactly, that’s what I proposed in my previous comment with the Fire and Forget mode. When using the dispatcher, it achieves exactly that: the child workflow execution instruction gets sent to a queue, which then gets picked up by any node in the cluster :)

**Comment by johnwc** [comment]
@sfmskywalker can you let us know when this new functionality has been added to the engine?

**Comment by Adrian-Grimm** [comment]
A pull of us got merged: https://github.com/elsa-workflows/elsa-core/pull/3455 
with this we have such a feature now.

**Comment by peter-mccann** [comment]
@johnwc I have a similar need to truly fire and forget due to multi hour build times. I'm unable to figure out how to set DispatchAndForget thru the APIs or in the workflow.  If you can provide a hint I would be grateful.
Thank you

**Comment by johnwc** [comment]
> @johnwc I have a similar need to truly fire and forget due to multi hour build times. I'm unable to figure out how to set DispatchAndForget thru the APIs or in the workflow. If you can provide a hint I would be grateful. Thank you

@peter-mccann Unfortunately, we had to move to using HangFire to run everything. We ended up migrating all our custom activities to HangFire jobs. We implemented managing state for the "flow" within a session table, it includes fields to track the session like... type of session it is, what stage it is on, what state the stage currently is in (active, error, done, etc...), retry counts. Each job runs and checks to see if there are any sessions with previous stage set as done, updates the stage and state to active and starts doing the work. With those basic things added as a base abstract class that all jobs inherit from; it was very simple to mimic basic flow. We would have loved to keep using Elsa, but we couldn't wait for the things we required. We still have it planned to revisit using Elsa when v3 matures a bit more, as we believe most of the pain points are going to be addressed in v3.
